OCPBUGS#18817: Add catalog health requirements (#64601)
openshift/openshift-docs: modules/olm-cs-health.adoc

// Module included in the following assemblies:
//
// * operators/understanding/olm/olm-understanding-olm.adoc

[id="olm-cs-health_{context}"]
= Catalog health requirements

Operator catalogs on a cluster are interchangeable from the perspective of installation resolution; a `Subscription` object might reference a specific catalog, but dependencies are resolved using all catalogs on the cluster.

For example, if Catalog A is unhealthy, a subscription referencing Catalog A could resolve a dependency in Catalog B, which the cluster administrator might not have been expecting, because B normally had a lower catalog priority than A.

> As a result, OLM requires that all catalogs on the cluster are healthy. When a catalog is unhealthy, all Operator installation or update operations will fail with a `CatalogSourcesUnhealthy` condition. If these operations were permitted in an unhealthy state, OLM might make resolution and installation decisions that were unexpected to the cluster administrator.

Reviewer: Actually, only the broken catalog source in the global namespace `openshift-marketplace` will block all cataglog sources. If the broken catalog source is in other namespaces, it will only block the subscription referring to the `catalogsource` of the same namespace, not others.

> As a result, OLM requires that all catalogs with a given global namespace (for example, the default `operator-marketplace` namespace or a custom global namespace) are healthy. When a catalog is unhealthy, all Operator installation or update operations within its shared global namespace will fail with a `CatalogSourcesUnhealthy` condition. If these operations were permitted in an unhealthy state, OLM might make resolution and installation decisions that were unexpected to the cluster administrator.

Reviewer: The default global namespace is `openshift-marketplace`, not `operator-marketplace`.

> As a cluster administrator, if you observe an unhealthy catalog and want to consider the catalog as invalid and resume Operator installations, see the "Removing custom catalogs" or "Disabling the default OperatorHub catalog sources" sections for information on removing the unhealthy catalog.

Reviewer 2: "...sections for information about removing the unhealthy catalog." Use "about" rather than "on".

// Module included in the following assemblies:
//
// * operators/understanding/olm/olm-understanding-dependency-resolution.adoc
// * operators/understanding/olm/olm-understanding-olm.adoc

[id="olm-cs-health_{context}"]
= Catalog health requirements

Operator catalogs on a cluster are interchangeable from the perspective of installation resolution; a `Subscription` object might reference a specific catalog, but dependencies are resolved using all catalogs on the cluster.

For example, if Catalog A is unhealthy, a subscription referencing Catalog A could resolve a dependency in Catalog B, which the cluster administrator might not have been expecting, because B normally had a lower catalog priority than A.

> As a result, OLM requires all catalogs on the cluster to be healthy; when a catalog is unhealthy, all Operator installation or update operations will fail with a `CatalogSourcesUnhealthy` condition. Otherwise, if these operations were permitted in this state, OLM might make resolution and installation decisions that were unexpected to the cluster administrator.

Reviewer 3: ```suggestion
As a result, OLM requires that all catalogs on the cluster are healthy. When a catalog is unhealthy, all Operator installation or update operations will fail with a `CatalogSourcesUnhealthy` condition. If these operations were permitted in an unhealthy state, OLM might make resolution and installation decisions that were unexpected to the cluster administrator.
```
